### DISCUSSION TITLE
Orca's jpg thumbnail preview not displaying in Professional Firmware

### DISCUSSION THREAD

**Discussion by Ergonomicmike** [mriscoc/Ender3V2S1#1184] [General]
I am using Orca Slicer v1.8.1. I have set the thumbnail size to 230x180, JPEG. But Professional Software reports "No Thumbnail."

(Ender 3 v2, DACAI.)

Is there a trick/hack that I can use in Orca to make the thumbnail display in PF?

**Comment by Ergonomicmike** [comment]
I found an answer. It's from
https://github.com/mriscoc/Ender3V2S1/issues/1176

When I change the capitalization of JPG in the thumbnail code generated by Orca to lower case, then the thumbnail displays in PF.

**Comment by mriscoc** [comment]
Just follow the instructions posted here: https://github.com/mriscoc/Ender3V2S1/wiki/How-to-generate-a-gcode-preview

**Comment by Ergonomicmike** [comment]
Thanks. I had used those instructions to make thumbnails work in Cura. It didn't occur to me that the same instructions contained instructions for Orca. (Since Orca offers thumbnail generation internally. (That is, I didn't think that Orca needed a separate script.))

**Comment by Ergonomicmike** [comment]
I am now using Giadej's special executable to generate the thumbnail in Orca. (180x180 PNG, per the updated instructions here in the wiki.)

But it would be easier for users to simply use the built in jpg thumbnail generator in Orca. 

Of course, "No job is impossible for the man who doesn't have to do it himself." And I know nothing about code. (Except FORTRAN.) So maybe it's not possible - or easy - to allow PF to recognize Upper Case JPG, in addition to small case jpg, in the gcode?

Or maybe someone can write a small post-processing script that searches for the Upper Case JPG in the gcode and changes it to lower case? (Hmm ... maybe I shoudl see if ChatGPT can script it?) Although that still involves finding, downloading, and installing a script in Orca.

**Reply by mriscoc** [reply]
The script currently changes the JPG string if it is found. Additionally, the script incorporates some information to be shown in the screen along with the thumbnail preview.

**Reply by Ergonomicmike** [reply]
So then I should use the script instead of  Giadej's executable?

**Reply by mriscoc** [reply]
@Giadej will update the executable soon, but you can use the script file now.

**Reply by Ergonomicmike** [reply]
Gracias.
